python/pipeline/utils/DLC_tools.py

```diff
@@ -0,0 +1,1033 @@
+
+import os
+# disable DLC GUI
+os.environ["DLClight"] = "True"
+
+import math
+import yaml
+import cv2
+from pathlib import Path
+import numpy as np
+import pandas as pd
+import ruamel.yaml
+import imageio
+import time
+
+from deeplabcut.utils import auxiliaryfunctions
+from deeplabcut.utils import video_processor
+from deeplabcut.utils import plotting
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+
+
+import pylab as pl
+from IPython import display
+import matplotlib.pyplot as plt
+
+
+def key_dict_generater(case):
+    case_key = {'animal_id': None, 'session': None, 'scan_idx': None}
+    for ind, key in enumerate(case_key.keys()):
+        case_key[key] = int(case.split('_')[ind])
+
+    return case_key
+
+
+def make_circumcircle(a, b, c):
+    # Mathematical algorithm from Wikipedia: Circumscribed circle
+    ox = (min(a[0], b[0], c[0]) + max(a[0], b[0], c[0])) / 2.0
+    oy = (min(a[1], b[1], c[1]) + max(a[1], b[1], c[1])) / 2.0
+    ax = a[0] - ox
+    ay = a[1] - oy
+    bx = b[0] - ox
+    by = b[1] - oy
+    cx = c[0] - ox
+    cy = c[1] - oy
+    d = (ax * (by - cy) + bx * (cy - ay) + cx * (ay - by)) * 2.0
+    if d == 0.0:
+        return None
+    x = ox + ((ax*ax + ay*ay) * (by - cy) + (bx*bx + by*by)
+              * (cy - ay) + (cx*cx + cy*cy) * (ay - by)) / d
+    y = oy + ((ax*ax + ay*ay) * (cx - bx) + (bx*bx + by*by)
+              * (ax - cx) + (cx*cx + cy*cy) * (bx - ax)) / d
+    ra = math.hypot(x - a[0], y - a[1])
+    rb = math.hypot(x - b[0], y - b[1])
+    rc = math.hypot(x - c[0], y - c[1])
+    return (x, y, max(ra, rb, rc))
+
+
+_MULTIPLICATIVE_EPSILON = 1 + 1e-14
+
+
+def is_in_circle(c, p):
+    return c is not None and math.hypot(p[0] - c[0], p[1] - c[1]) <= c[2] * _MULTIPLICATIVE_EPSILON
+
+
+def smallest_enclosing_circle_naive(points):
+    # Degenerate cases
+    if len(points) == 0:
+        return None
+    elif len(points) == 1:
+        return (points[0][0], points[0][1], 0)
+
+    result = None
+    # Try all unique triples
+    for i in range(len(points)):
+        p = points[i]
+        for j in range(i + 1, len(points)):
+            q = points[j]
+            for k in range(j + 1, len(points)):
+                r = points[k]
+                c = make_circumcircle(p, q, r)
+                if c is not None and (result is None or c[2] < result[2]) and \
+                        all(is_in_circle(c, s) for s in points):
+                    result = c
+
+    if result is None:
+        raise AssertionError()
+    return result
+
+class PlotBodyparts():
+
+    def __init__(self, config, bodyparts='all'):
+        """
+        Input:
```

Comment:
the standard docstring uses Arguments: and Returns: rather than Input and Output (like here https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html). You don't need to change it, just for future reference

Reply:
Got it.